src/msg/async/dpdk/DPDKStack.h

```diff
@@ -137,10 +137,9 @@ class NativeConnectedSocketImpl : public ConnectedSocketImpl {
 
     std::vector<fragment> frags;
     auto pb = bl.buffers().begin();
-    uint64_t left_pbrs = bl.buffers().size();
```

Comment:
i'd suggest put
```c++
uint64_t left_pbrs = bl.get_num_buffers();
```
to avoid creating a temporary iterator in each iteration of the `while` loop.

Reply:
Not quite follow you. Pb anyway is used as iterator, not sure what do you mean a temp iterator is created.

Reply:
in your change,

1. `bl.buffers().end()` returns a temporary object, and
1. it is compared with `pb`, then
1. it is dropped on the floor.

Reply:
const buffers_t& buffers() const { return _buffers; }
 const_iterator buffers_t::end() const {
	return &_root;
  }

You can see both function calls returns const variables and no new object created. And this is very typical pattern in C++ to enum over a list by a iterator.

Reply:
i see. thought it was `bl.end()`.